File: AndroidApp/app/src/main/java/com/tv2oppgave/tvmovieapp/data/models/movie.kt
package com.tv2oppgave.tvmovieapp.data.models

data class MovieDetailResponse(
    val id: String? = null,
    val title: String? = null,
    val description: String? = null,
    val durationMinutes: Int? = null,
    val posterUrl: String? = null,
    val year: Int? = null,
    val ageRating: String? = null,
    val genres: List<String> = emptyList(),
    val cast: List<String> = emptyList()
)
data class MovieImages(
    val poster: MovieImageVariant? = null,
    val cover: MovieImageVariant? = null,
    val thumbs: List<MovieImageVariant>? = null
)

data class MovieImageVariant(
    val src: String? = null,
    val width: Int? = null,
    val height: Int? = null
)

File: WebApp/src/models/movie.ts
import { Label } from "./feed";

export interface MovieDetail {
  id: string;
  title: string;
  description: string;
  durationMinutes: number | null;
  posterUrl: string | null;
  year: number | null;
  ageRating: string | null;
  genres: string[];
  cast: string[];
}
